aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -1030,22 +1032,27 @@ async def apply_requirements(
             for descriptor_id in result.keys():
                 credential_list = result.get(descriptor_id)
                 for credential in credential_list:
-                    if credential.given_id not in given_id_descriptors:
-                        given_id_descriptors[credential.given_id] = {}
-                    given_id_descriptors[credential.given_id][descriptor_id] = {}
+                    if credential.given_id:
+                        if credential.given_id not in cred_uid_descriptors:
+                            cred_uid_descriptors[credential.given_id] = {}
+                        cred_uid_descriptors[credential.given_id][descriptor_id] = {}
+                    else:
+                        if credential.record_id not in cred_uid_descriptors:
+                            cred_uid_descriptors[credential.record_id] = {}
+                        cred_uid_descriptors[credential.record_id][descriptor_id] = {}
```

Comment:
This can be simplified a bit, maybe something like:

```py
cred_id = credential.given_id or credential.record_id
if cred_id:
  cred_uid_descriptors.setdefault(cred_id, {})[descriptor_id] = {}
```

Reply:
Simplified using the above template in a few places.